data-collection/src/main/java/refactoringml/PMDatabase.java

```diff
@@ -39,11 +38,14 @@ In case of (various renames), the names are replaced, e.g.
 	 */
 	public ProcessMetricTracker renameFile(String oldFileName, String newFileName, CommitMetaData commitMetaData){
 		ProcessMetricTracker pmTracker = new ProcessMetricTracker(database.getOrDefault(oldFileName, new ProcessMetricTracker(newFileName, commitMetaData)));
-		pmTracker.setFileName(newFileName);
-		//the class file was renamed or moved, but the refactoring was not detected by refactoring miner
-		if(pmTracker.getCommitCounter() != 0){
-			pmTracker.resetCounter(commitMetaData);
+		if(oldFileName.equals(newFileName)){
```

Comment:
I don't get these `if`s here. In which cases we would call this `renameFile` method and 1) the old file doesn't exist in the database?, or 2) the old and new file names would be the same?

Moreover, the `if(database.get(newFileName) != null)`? Why do we need it?

Can you add a few comments in this method?

Reply:
Both checks cover cases in which the rename refactoring is incorrect.
The first checks if at least some kind of rename occurred.
The second one checks if a rename onto an already existing file occurred.

Both cases are failures of our implementation, thus I changed them to throw Exceptions.